source/common/grpc/async_client_impl.cc

```diff
@@ -60,6 +60,15 @@ AsyncStreamImpl::AsyncStreamImpl(AsyncClientImpl& parent,
     : parent_(parent), service_method_(service_method), callbacks_(callbacks), timeout_(timeout) {}
 
 void AsyncStreamImpl::initialize(bool buffer_body_for_retry) {
+  auto clusters = parent_.cm_.clusters();
```

Comment:
Don't make a copy of clusters here, it could be expensive.

Just use .get(cluster_name) to check if you're getting nullptr.